### ISSUE TITLE
update needed. 

### ISSUE THREAD

**Issue by DJ4everkill** [Lacyway/V-Rising-Server-Manager#23]
latest update have added "frames when empty" and a ton more settings both for host and game settings. 
don't know how either but the discord webhook stopped working, just now. was working earlier today when the game got an update. 

an option for message when the server status is online and joinable. could be when the serverlog sends this message "Shutting down Asynchronous Streaming" i havn't see it anywere els. could be wrong. but you know what i mean :D

PS. Nice work, keep it up :D

**Comment by Lacyway** [comment]
Thank you!

I am about 75% done with a semi-big update, I'll have a look at all this as well. The webhook issue sounds strange but I might have an idea.

Should hopefully be done with the update this week.

**Comment by DJ4everkill** [comment]
oh. found out they get SUPER delayed. shut down server around 01:51 and a message came up around 04:50.
started the server this morning around 09:21 and havn't still got anything on the Discord. 
before i got message almost instant, but after the update and update in 5 min message, it got super delayed. they came 1/6-23 11:50, don't know if it's because they both got send at the same time?

**Comment by Lacyway** [comment]
If you are still running a server, could you see if the latest release has this issue? Thanks. 😊

**Comment by DJ4everkill** [comment]
after a day or two the issue stopped, and messages came through as soon i pressed the button. so may have been a discord server issue. 
testing new release now :D

**Comment by Lacyway** [comment]
Thank you.

I tried making it as easy as possible to use. If you have the time, please let me know what you think in comparison to the old version. If you are migrating your server to the new version, _**don't forget to create backups**_.

The remove server is currently bugged but I'll release a new version tomorrow fixing it.

**Comment by DJ4everkill** [comment]
when i try to update a server with any other name in folder it sends
Failed to load script file 'C:\V-Rising' (real server path--> 'C:\V-Rising Server') might not be able to read (space) might just need som "serverpath" quotes in code. 

when failing updating server due to space in server path, it comes with error code, and is just stuck on updating. people with no logic to look in steamcmd window, is lost and will just wait for update, maybe even uninstall again. 

could remove the server tab without issue. no idéa what is bugged?
no webhook edit messages, i need those to give the messages alittle spice. also to remove the ip. 

should load game and host settings in server folder save. C:\*\SaveData\Settings
so maybe option to save elswere, or just press save and it saves in that location. 

bigger "hitbox" for settings button, where you have to hit is very small :D 

RCON console is not available. it's activated in settings. 

maybe add the "Add server" button up right at the server fan list. together with VSM settings makes more sense up there. 

option for different webhook messages for each server. would be very handy. 

thats everything my eye hit right away :D
look awesome with new overlay. maybe add an darkmode (love that)

**Comment by Lacyway** [comment]
Big thanks!

I will have a look at all this tomorrow. RCON is temporarily disabled but should be back soon. I'll re-add the discord messages again, it's already planned, and per server is a good idea. The "Add Server" thing is hard to implement, I've been thinking about it but haven't found a way to do it "elegantly".

Only thing I'm wondering about is the Settings button, do you mean VSM Settings? The hitbox should act normal so it sounds a bit weird.

Thank you again!

**Comment by DJ4everkill** [comment]
no the hitboxes in host settings and game settings files. 

and elegant maybe like this, and the +moves with server fans :D 
![billede](https://github.com/Lacyway/V-Rising-Server-Manager/assets/11280026/50fecdd6-83c0-434b-a77b-e3834cdb66f9)

**Comment by Lacyway** [comment]
Ah, I understand!

If you mean the ones with digits specifically (with up and down arrows on the side) they are meant to be replaced in the future, I did notice myself that they were frustrating to click on. I haven't gotten that far yet.

I'll see what I can do with the main UI in regards to your image. Thank you again for taking the time, I appreciate the feedback. 😊

**Comment by DJ4everkill** [comment]
no problem 😄  it's nice to have a little "touch" on the work 😄  especially when i use it my self 😄 not much of a coder my self tho, only basic understanding 😅

**Comment by Lacyway** [comment]
I've fixed almost everything above now.
- Path can now contains spaces
- Dark mode works
- Game and Save Settings Editors now asks if you wanna save to a server and will automatically create it at the right place
- All webhook messages are now customizable
- RCON has been added again (a bit ugly, will improve over time)
- Hitboxes fixed, all number boxes now use another control which looks and "acts" better

To do:
- Add "New Server" button on the tabcontrol (might push the release before I can add this, haven't found a good method)

Need a bit more time then I should be able to push the release. Sorry for the wait.

**Comment by Lacyway** [comment]
I pushed the new release now, you can download it [here](https://github.com/Lacyway/V-Rising-Server-Manager/releases/tag/v3.2-beta) or use the auto-updater. Let me know if it's like you wanted. Still trying to figure out where to put "Settings" and the "New" button for servers.

**Comment by DJ4everkill** [comment]
awesome work :D love the result :D could you add auto load of host and game settings? and maybe make it depend on what server fan you are on? if not on the fan it's fine with the current dropdown menu when you save :D but at least so i don't have to look after the files them self.

**Comment by Lacyway** [comment]
Of course. I'll make it a setting that you toggle, if enabled it will autoload the selected server when you open the editor (if it exists). I'll have a look at it this weekend. Let me know if you have any other suggestions or find any bugs.

**Comment by Lacyway** [comment]
Auto loading has been added in the latest release today.

**Comment by DJ4everkill** [comment]
so nice to see :D hard to see now how it would be any better :D maybe option for modinstall and auto update on that part. 
only thingis, only dev builds of mods is available atm, cause the gloomrot update smashed the mods. 
maybe if you are up for it, create a mod that will pull the player count, maybe even playerlist. so you can see if the server is empty or have players onlne :D 

somthing completly diffrent, but hey if you have the time and want to explore i would say go you it :D

**Comment by Lacyway** [comment]
I've made some progress and should be able to release a new version with mod management being included. Hopefully it'll be done by this weekend. 👍 
![image](https://github.com/Lacyway/V-Rising-Server-Manager/assets/20912169/1d9534da-d673-44c3-b525-1511d966228f)

**Comment by DJ4everkill** [comment]
looks awesome. but just encounted a bug. when the server auto updates it closes the server but the auto restart starts the server while updating the server, making the application come to either alot of booting messages and fail, somtimes even crash. 

the auto restart server needs to have a 5-10 sec delay, check if steamCMD is running and still updating, then wait.

**Comment by Lacyway** [comment]
Hmm, I've thoroughly tested that and it shouldn't happen. I'll have a look at it.

**Comment by DJ4everkill** [comment]
my server does somthing els XD 
![billede](https://github.com/Lacyway/V-Rising-Server-Manager/assets/11280026/fb2c73ef-6081-4246-8fdf-a86c8e9d10a6)

**Comment by DJ4everkill** [comment]
the one from 16:08 is manuel reboot

**Comment by Lacyway** [comment]
I found the error, I will add it to the next release.

**Comment by Lacyway** [comment]
The new version is getting flagged by anti viruses again, haven't had this happen since last year. I'll have to revert the latest version and try to fix it...

**Comment by Lacyway** [comment]
Should be uploaded now, let me know if it works. I think the mod manager is triggering some false detections.

**Comment by DJ4everkill** [comment]
Looks good. If you could implement the tags from the store, that would be really helpful, cause many plugins is outdated, and some is only for clients and some for only servers. Newst plugins is tagged with gloomrot update tag, but i have seen some updated without the tag. Not easy to go around.

**Comment by DJ4everkill** [comment]
woop :D new feedback :D leave an check off for webhook message per server. having a test server running, don't need messages to be sent, and have custom messages set and ready for when i need them. so deleting them would be a hassel :D

**Comment by Lacyway** [comment]
Sure, I'll add that. :)

**Comment by Lacyway** [comment]
I'm gonna close this issue because of its length, if you find any bugs or new feature requests please let me know in a new one. And thank you again for all the support.
